zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -28,19 +37,92 @@ impl TryFrom<&Transaction> for zcash_primitives::transaction::Transaction {
             | Transaction::V4 { .. } => panic!("Zebra only uses librustzcash for V5 transactions"),
         };
 
-        let serialized_tx = trans.zcash_serialize_to_vec()?;
-        // The `read` method currently ignores the BranchId for V5 transactions;
-        // but we use the correct BranchId anyway.
-        let branch_id: u32 = network_upgrade
-            .branch_id()
-            .expect("Network upgrade must have a Branch ID")
-            .into();
-        // We've already parsed this transaction, so its network upgrade must be valid.
-        let branch_id: zcash_primitives::consensus::BranchId = branch_id
-            .try_into()
-            .expect("zcash_primitives and Zebra have the same branch ids");
-        let alt_tx =
-            zcash_primitives::transaction::Transaction::read(&serialized_tx[..], branch_id)?;
-        Ok(alt_tx)
+        convert_tx_to_librustzcash(trans, *network_upgrade)
+    }
+}
+
+fn convert_tx_to_librustzcash(
+    trans: &Transaction,
+    network_upgrade: NetworkUpgrade,
+) -> Result<zcash_primitives::transaction::Transaction, io::Error> {
+    let serialized_tx = trans.zcash_serialize_to_vec()?;
+    let branch_id: u32 = network_upgrade
+        .branch_id()
+        .expect("Network upgrade must have a Branch ID")
+        .into();
+    // We've already parsed this transaction, so its network upgrade must be valid.
+    let branch_id: zcash_primitives::consensus::BranchId = branch_id
+        .try_into()
+        .expect("zcash_primitives and Zebra have the same branch ids");
+    let alt_tx = zcash_primitives::transaction::Transaction::read(&serialized_tx[..], branch_id)?;
+    Ok(alt_tx)
+}
```

Comment:
Could this be a method on `Transaction` such as `Transaction::into_librustzcash(network_upgrade: NetworkUpgrade) -> Result<zcash_primitives::transaction::Transaction, io::Error>` or something?  I'm thinking of: https://rust-lang.github.io/api-guidelines/naming.html#c-conv

Reply:
Oh! I totally missed that. You are right, ignore me.